server/src/Validation.hs

```diff
@@ -142,11 +142,11 @@ noMatches filters =
 
 -- | Ensure a a Customer with the given email doesn't exist.
```

Comment:
```suggestion
-- | Ensure a registered (i.e., non-ephemeral) Customer with the given email doesn't exist.
```
Since the logic of the function has changed, and it actually allows the same email to be used for an ephemeral account